vendor/golang.org/x/sys/unix/syscall_darwin_arm64.go

```diff
@@ -47,5 +47,6 @@ func Syscall9(num, a1, a2, a3, a4, a5, a6, a7, a8, a9 uintptr) (r1, r2 uintptr,
 //sys	getfsstat(buf unsafe.Pointer, size uintptr, flags int) (n int, err error) = SYS_GETFSSTAT
 //sys	Lstat(path string, stat *Stat_t) (err error)
 //sys	ptrace1(request int, pid int, addr uintptr, data uintptr) (err error) = SYS_ptrace
+//sys	ptrace1Ptr(request int, pid int, addr unsafe.Pointer, data uintptr) (err error) = SYS_ptrace
 //sys	Stat(path string, stat *Stat_t) (err error)
 //sys	Statfs(path string, stat *Statfs_t) (err error)
```

Comment:
The patch adds a new function `ptrace1Ptr` that takes an unsafe pointer as an argument instead of `uintptr`. This could potentially introduce bugs if the pointer is not properly handled or dereferenced. It would be safer to pass the pointer as a uintptr and use the unsafe package to convert it back to a pointer within the function.

Additionally, it might be useful to add some comments or documentation to the new function explaining its purpose and usage.